src/scripts/main.js

```diff
@@ -1,3 +1,317 @@
+      row.appendChild(cell);
+    });
+    tableBody.appendChild(row);
+  });
+}
+
+function toggleSortDirection() {
+  sortDirection = !sortDirection;
+}
+
+function attachSortingListeners() {
+  table.querySelectorAll('th').forEach((header, index) => {
+    header.addEventListener('click', () => sortTable(index));
+  });
+}
+
+attachSortingListeners();
+
+// 2. When user clicks on a row, it should become selected.
+let activeRow = null;
+const tbody = table.querySelector('tbody');
+
+tbody.addEventListener('click', (e) => {
+  const row = e.target.closest('tr');
+
+  if (!row) {
+    return;
+  }
+
+  if (activeRow) {
+    activeRow.classList.remove('active');
+  }
+
+  row.classList.add('active');
+  activeRow = row;
+});
+
+// 3. Write a script to add a form to the document.
+// Form allows users to add new employees to the spreadsheet.
+const formHtml = `
+  <form class="new-employee-form">
+    <label>
+      Name:
+      <input
+        name="name"
+        type="text"
+        data-qa="name"
+      />
+    </label>
+    <label>
+      Position:
+      <input
+        name="position"
+        type="text"
+        data-qa="position"
+      />
+    </label>
+    <label>
+      Office:
+      <select
+        name="office"
+        data-qa="office"
+      >
+        <option value="Tokyo">Tokyo</option>
+        <option value="Singapore">Singapore</option>
+        <option value="London">London</option>
+        <option value="New York">New York</option>
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <label>
+      Age:
+      <input
+        name="age"
+        type="number"
+        data-qa="age"
+      />
+    </label>
+    <label>
+      Salary:
+      <input
+        name="salary"
+        type="number"
+        data-qa="salary"
+      />
+    </label>
+    <button
+      type="button"
+      id="save-button"
+    >
+      Save to table
+    </button>
+  </form>
+  `;
+
+table.insertAdjacentHTML('afterend', formHtml);
+
+// 4. Show notification if form data is invalid
+function pushNotification(message, type) {
+  const notification = document.createElement('div');
+  const titleSpan = document.createElement('span');
+
+  notification.className = `notification ${type}`;
+  notification.setAttribute('data-qa', 'notification');
+
+  titleSpan.className = 'title';
+  titleSpan.textContent = type.toUpperCase();
+
+  notification.appendChild(titleSpan);
+  notification.innerHTML += ` <span>${message}</span>`;
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 2000);
+}
+
+function validateName(formName) {
+  const nameRegex = /^[A-Za-z\s]+$/;
+
+  if (formName.length < 4 || !nameRegex.test(formName)) {
+    pushNotification(
+      'Name must be at least 4 characters long and contain only letters.',
+      'error',
+    );
+
+    return false;
+  }
+
+  return true;
+}
```

Comment:
Code quality issue: The `validateName` function is doing two things: it's validating the name and displaying an error message if the validation fails. It would be better to separate these concerns. Have one function to validate the data and another function to handle the display of error messages.